---
title: "EWAS for immune function genes"
author: "Ji-Qing Chen"
date: "5/9/2022"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(dplyr)
library(xlsx)
library(IlluminaHumanMethylationEPICanno.ilm10b4.hg19)
library(tidyverse)
'%!in%' <- Negate('%in%')

library(survival)
library(survminer)
```

# Load Imputed beta value
```{r}
beta <- readRDS("/Users/chenjiqing/Public/bladder cancer/raw data/AllBC_Combat_Betas.rds")
anyNA(beta)
dim(beta)

annot <- readRDS("/Users/chenjiqing/Public/bladder cancer/GSEA and LOLA and GenomicContext/EPIC.hg19.manifest.rds")
annot<- as.data.frame(annot) %>% rownames_to_column(.)
table(annot$MASK_general)
row_contain_rs <- annot %>% filter(grepl("rs", .$rowname))
row_contain_ch <- annot %>% filter(grepl("ch", .$rowname))
annot_filtered <- annot %>% 
                  filter(rowname %!in% row_contain_rs$rowname) %>%
                  filter(rowname %!in% row_contain_ch$rowname) %>%
                  filter(MASK_general == "FALSE") %>%
                  filter(seqnames != "chrX") %>% 
                  filter(seqnames != "chrY")
rm(row_contain_rs,row_contain_ch)

beta2 <- as.data.frame(beta) %>% rownames_to_column(.)
rm(beta,annot)
beta.filtered <- beta2 %>% 
                 filter(rowname %in% annot_filtered$rowname)
rm(beta2,annot_filtered)
beta.filtered <- beta.filtered %>% column_to_rownames(., var = "rowname") %>% as.matrix(.)
dim(beta.filtered) # 20,891 CpGs
```

# Get NMIBC patients betas
```{r}
load("/Users/chenjiqing/Public/bladder cancer/Basic Analysis/data_clean2.RData")
rm(phenotype1,Recu)
NMIBC <- phenotype2 %>% 
         filter(muscinv =="no")

# weird data
weird = phenotype2[which(phenotype2$ddstat=="dead" & phenotype2$RecurYN=="yes" &
                         phenotype2$survreftmd<phenotype2$recsurgtmd),]
NMIBC <- NMIBC %>% filter(Sample_Name != weird$Sample_Name)

a <- as.character(NMIBC$Sample_Name)
NMIBC.betas_immune <- beta.filtered[,a]
dim(NMIBC.betas_immune)

rm(beta.filtered,weird,phenotype2)
```

# load phenotype
```{r}
# cell proportion
load("/Users/chenjiqing/Public/bladder cancer/Extended Deconvolution/Bladder_Extended_cell_proportion.RData")

# remove weird data
NMIBC_immune <- NMIBC %>%
              mutate(Age = refage, Sex = sex, Tumor_grade = grade2, 
                     Smoking_status = smkstat2, BCG = ImmunoTx) %>%
              dplyr::select(Sample_Name,Age,Sex,Tumor_grade,Smoking_status,
                            BCG,TenDead, TenYearSurv, TenRFS, TenYearRFS) %>%
              left_join(Extended2) %>%
              mutate(T_cell = CD4mem + CD4nv + CD8mem + CD8nv + Treg, # not show this one
                     Lymph = Bmem + Bnv + NK + T_cell) %>%
              mutate(NLR = ifelse(Lymph == 0,Neu/0.01, Neu/Lymph)) %>%
              dplyr::select(-T_cell,-Lymph)

# Winsorization
NMIBC_immune_win <- NMIBC_immune %>%
         mutate(Bas = ifelse(Bas <= quantile(NMIBC_immune$Bas, probs = 0.98),
                             Bas,quantile(NMIBC_immune$Bas, probs = 0.98)),
                Bmem = ifelse(Bmem <= quantile(NMIBC_immune$Bmem, probs = 0.98),
                              Bmem,quantile(NMIBC_immune$Bmem, probs = 0.98)),
                Bnv = ifelse(Bnv <= quantile(NMIBC_immune$Bnv, probs = 0.98),
                             Bnv,quantile(NMIBC_immune$Bnv, probs = 0.98)),
                CD4mem = ifelse(CD4mem <= quantile(NMIBC_immune$CD4mem, probs = 0.98),
                                CD4mem,quantile(NMIBC_immune$CD4mem, probs = 0.98)),
                CD4nv = ifelse(CD4nv <= quantile(NMIBC_immune$CD4nv, probs = 0.98),
                               CD4nv,quantile(NMIBC_immune$CD4nv, probs = 0.98)),
                CD8mem = ifelse(CD8mem <= quantile(NMIBC_immune$CD8mem, probs = 0.98),
                                CD8mem,quantile(NMIBC_immune$CD8mem, probs = 0.98)),
                CD8nv = ifelse(CD8nv <= quantile(NMIBC_immune$CD8nv, probs = 0.98),
                               CD8nv,quantile(NMIBC_immune$CD8nv, probs = 0.98)),
                Eos = ifelse(Eos <= quantile(NMIBC_immune$Eos, probs = 0.98),
                             Eos,quantile(NMIBC_immune$Eos, probs = 0.98)),
                Mono = ifelse(Mono <= quantile(NMIBC_immune$Mono, probs = 0.98),
                              Mono,quantile(NMIBC_immune$Mono, probs = 0.98)),
                Neu = ifelse(Neu <= quantile(NMIBC_immune$Neu, probs = 0.02),
                             quantile(NMIBC_immune$Neu, probs = 0.02),Neu),
                NK = ifelse(NK <= quantile(NMIBC_immune$NK, probs = 0.98),
                            NK,quantile(NMIBC_immune$NK, probs = 0.98)),
                Treg = ifelse(Treg <= quantile(NMIBC_immune$Treg, probs = 0.98),
                              Treg,quantile(NMIBC_immune$Treg, probs = 0.98)),
                NLR = ifelse(NLR <= quantile(NMIBC_immune$NLR, probs = 0.98),
                             NLR,quantile(NMIBC_immune$NLR, probs = 0.98)))
rm(NMIBC,Extended2)
```

# Get annotation
```{r}
annEPIC <- getAnnotation("IlluminaHumanMethylationEPICanno.ilm10b4.hg19")
```

# EWAS 
```{r}
mVals <- minfi::logit2(NMIBC.betas_immune)
# save for autoencoder kmeans
#mVals2 <- t(mVals)
#View(head(mVals2))
#write.csv(mVals2, file = "/Users/chenjiqing/Public/bladder cancer/Extended Deconvolution/Script/Autoencoder_Kmeans/mVals_20891.csv",row.names = F)

annEPICSub <- annEPIC[match(rownames(mVals),annEPIC$Name),]

# Cox EWAS
source("/Users/chenjiqing/Public/bladder cancer/EWAS/ewaff_for_cox.R")

library(rms)
os <- coxph(Surv(TenYearSurv, TenDead) ~ Age + Sex + Tumor_grade + Smoking_status + BCG + Bas + Bmem + Bnv + CD4mem + CD4nv + CD8mem + CD8nv + Neu + NK + Mono + Treg + Eos, data=NMIBC_immune_win)
vif(os) # Neu colinearity

DMPs_cell_cox_OS <- ewaff.sites(formula = Surv(TenYearSurv, TenDead) ~ Age + Sex + Tumor_grade + Smoking_status + BCG + Bas + Bmem + Bnv + CD4mem + CD4nv + CD8mem + CD8nv + NK + Mono + Treg + Eos,
                    methylation = mVals,
                    data = NMIBC_immune_win,
                    method = "coxph",
                    generate.confounders = NULL)
View(DMPs_cell_cox_OS$table)

os <- coxph(Surv(TenYearRFS, TenRFS) ~ Age + Sex + Tumor_grade + Smoking_status + BCG + Bas + Bmem + Bnv + CD4mem + CD4nv + CD8mem + CD8nv + Neu + NK + Mono + Treg + Eos, data=NMIBC_immune_win)
vif(os) # Neu colinearity

DMPs_cell_cox_RFS <- ewaff.sites(formula = Surv(TenYearRFS, TenRFS) ~ Age + Sex + Tumor_grade + Smoking_status + BCG + Bas + Bmem + Bnv + CD4mem + CD4nv + CD8mem + CD8nv + NK + Mono + Treg + Eos,
                    methylation = mVals,
                    data = NMIBC_immune_win,
                    method = "coxph",
                    generate.confounders = NULL)
View(DMPs_cell_cox_RFS$table)


save(DMPs_cell_cox_OS,DMPs_cell_cox_RFS, file = "/Users/chenjiqing/Public/bladder cancer/Extended Deconvolution/Script/EWAS/EWAS_NMIBC_BothOutcomes.RData")
```

# volcano plot DMPs_cell_cox_OS
```{r}
with(DMPs_cell_cox_OS$table, plot(estimate, -log10(p.value), pch=20, xlab = "coefficient of CpG",ylab = "-log10(P value)", 
                                  #xlim=c(-1.2,1.2), 
                                  ylim = c(0,11),
                                  cex=0.5,bty="L")) +
with(subset(DMPs_cell_cox_OS$table, p.value<0.005 ), points(estimate, -log10(p.value), pch=20, col="red", cex=0.5))
legend(0,11, col = c("red"), legend = c("P value < 0.005"), pch = 20, cex = 0.7,box.lty=0)
text(-3,8.5,c("8,540 CpGs"),cex=0.7, col = c("red"))
text(1,8.5,c("1,011 CpGs"),cex=0.7, col = c("red"))
```

# Finding common CpGs (OS)
```{r}
load("/Users/chenjiqing/Public/bladder cancer/Extended Deconvolution/Script/EWAS/EWAS_NMIBC_BothOutcomes.RData")

OS_0.005 <- DMPs_cell_cox_OS$table %>% filter(p.value < 0.005) # 9551 CpGs

OS_0.005 <- OS_0.005 %>% rownames_to_column(var = "CpG")


# add annotation
annEPIC <- getAnnotation("IlluminaHumanMethylationEPICanno.ilm10b4.hg19")
annEPIC <- as.data.frame(annEPIC) %>% rownames_to_column(var = "CpG")
ann <- annEPIC %>% 
       select(CpG,Phantom5_Enhancers,DNase_Hypersensitivity_Evidence_Count,
              Relation_to_Island,chr,UCSC_RefGene_Name,UCSC_RefGene_Group) %>%
       mutate(DHS = ifelse(DNase_Hypersensitivity_Evidence_Count == "","No","Yes"),
              Enhancer = ifelse(Phantom5_Enhancers == "","No","Yes")) %>%
       select(CpG,DHS,Enhancer,Relation_to_Island,
              chr,UCSC_RefGene_Name,UCSC_RefGene_Group)

OS_0.005 <- OS_0.005 %>% left_join(., ann)

# Previous paper CpG
load("/Users/chenjiqing/Public/bladder cancer/EWAS/RFS/EWAS_strata_winsor.RData")

DMPs_0.005_cell <- DMPs_cell$table %>%
                   rownames_to_column(., var = "CpG") %>%
                   filter(p.value < 0.005)

DMPs_0.005_cell2 <- DMPs_0.005_cell %>%
                    select(CpG, estimate, p.value, p.adjust) %>%
                    left_join(.,ann)

# Find Common CpGs
common_CpG <- intersect(DMPs_0.005_cell2$CpG,OS_0.005$CpG)
common_CpG <- OS_0.005 %>% filter(CpG %in% common_CpG)

common_CpG = common_CpG[order(common_CpG$p.adjust,decreasing=F),]

write_csv(common_CpG ,"/Users/chenjiqing/Public/bladder cancer/Extended Deconvolution/Script/EWAS/OS_overlap_with_previousPaper_CpGs.csv")
```

# volcano plot DMPs_cell_cox_RFS
```{r}
with(DMPs_cell_cox_RFS$table, plot(estimate, -log10(p.value), pch=20, xlab = "coefficient of CpG",ylab = "-log10(P value)", 
                                  #xlim=c(-1.2,1.2), 
                                  ylim = c(0,7),
                                  cex=0.5,bty="L")) +
with(subset(DMPs_cell_cox_RFS$table, p.value<0.005 ), points(estimate, -log10(p.value), pch=20, col="red", cex=0.5))
legend(0,7, col = c("red"), legend = c("P value < 0.005"), pch = 20, cex = 0.7,box.lty=0)
text(-0.5,5.5,c("705 CpGs"),cex=0.7, col = c("red"))
text(0.5,5.5,c("737 CpGs"),cex=0.7, col = c("red"))
```

# Finding common CpGs (RFS)
```{r}
load("/Users/chenjiqing/Public/bladder cancer/Extended Deconvolution/Script/EWAS/EWAS_NMIBC_BothOutcomes.RData")

RFS_0.005 <- DMPs_cell_cox_RFS$table %>% filter(p.value < 0.005) # 9551 CpGs

RFS_0.005 <- RFS_0.005 %>% rownames_to_column(var = "CpG")


# add annotation
annEPIC <- getAnnotation("IlluminaHumanMethylationEPICanno.ilm10b4.hg19")
annEPIC <- as.data.frame(annEPIC) %>% rownames_to_column(var = "CpG")
ann <- annEPIC %>% 
       select(CpG,Phantom5_Enhancers,DNase_Hypersensitivity_Evidence_Count,
              Relation_to_Island,chr,UCSC_RefGene_Name,UCSC_RefGene_Group) %>%
       mutate(DHS = ifelse(DNase_Hypersensitivity_Evidence_Count == "","No","Yes"),
              Enhancer = ifelse(Phantom5_Enhancers == "","No","Yes")) %>%
       select(CpG,DHS,Enhancer,Relation_to_Island,
              chr,UCSC_RefGene_Name,UCSC_RefGene_Group)

RFS_0.005 <- RFS_0.005 %>% left_join(., ann)

# Previous paper CpG
load("/Users/chenjiqing/Public/bladder cancer/EWAS/RFS/EWAS_strata_winsor.RData")

DMPs_0.005_cell <- DMPs_cell$table %>%
                   rownames_to_column(., var = "CpG") %>%
                   filter(p.value < 0.005)

DMPs_0.005_cell2 <- DMPs_0.005_cell %>%
                    select(CpG, estimate, p.value, p.adjust) %>%
                    left_join(.,ann)

# Find Common CpGs
common_CpG <- intersect(DMPs_0.005_cell2$CpG,RFS_0.005$CpG)
common_CpG <- RFS_0.005 %>% filter(CpG %in% common_CpG)

common_CpG = common_CpG[order(common_CpG$p.adjust,decreasing=F),]

write_csv(common_CpG ,"/Users/chenjiqing/Public/bladder cancer/Extended Deconvolution/Script/EWAS/RFS_overlap_with_previousPaper_CpGs.csv")
```


# TNIP2
```{r}
library(sesame)

# beta
beta <- readRDS("/Users/chenjiqing/Public/bladder cancer/raw data/AllBC_Combat_Betas.rds")
# pheno
load("/Users/chenjiqing/Public/bladder cancer/Basic Analysis/data_clean2.RData")
rm(phenotype1,Recu)
NMIBC <- phenotype2 %>% 
         filter(muscinv =="no")

# weird data
weird = phenotype2[which(phenotype2$ddstat=="dead" & phenotype2$RecurYN=="yes" &
                         phenotype2$survreftmd<phenotype2$recsurgtmd),]
NMIBC <- NMIBC %>% filter(Sample_Name != weird$Sample_Name)

a <- as.character(NMIBC$Sample_Name)
NMIBC.betas <- beta[,a]

visualizeGene("TNIP2",NMIBC.betas,platform = "EPIC",refversion = "hg19",show.sampleNames=F,cluster.samples=T)
```

# cg02786370 survival
```{r}
library(survival)
library(survminer)
library(tidyverse)

# load beta
beta <- readRDS("/Users/chenjiqing/Public/bladder cancer/raw data/AllBC_Combat_Betas.rds")

# Select CpG site cg02786370
cg02786370 <- as.data.frame(beta["cg02786370",])

# Merge with imputed phenotype data
load("/Users/chenjiqing/Public/bladder cancer/Basic Analysis/data_clean2.RData")
rm(phenotype1,Recu)
NMIBC <- phenotype2 %>% 
         filter(muscinv =="no")

# weird data
weird = phenotype2[which(phenotype2$ddstat=="dead" & phenotype2$RecurYN=="yes" &
                         phenotype2$survreftmd<phenotype2$recsurgtmd),]
NMIBC <- NMIBC %>% filter(Sample_Name != weird$Sample_Name)

cg02786370 <- rownames_to_column(cg02786370)
colnames(cg02786370) <- c("Sample_Name", "cg02786370_beta")
NMIBC_cg02786370 <- left_join(NMIBC, cg02786370) %>%
               mutate(cg02786370_beta = ifelse( cg02786370_beta>median(cg02786370_beta), "high cg02786370 Beta","low cg02786370 Beta"))

# KM plot for OS
cg02786370_fit_10y = survfit(Surv(TenYearSurv, TenDead) ~ cg02786370_beta, data = NMIBC_cg02786370)
OS_cg10642330_10y <- ggsurvplot(
  cg02786370_fit_10y,
  xlab = "Months", 
  ylab = "10-year overall survival probability",
  ylim = c(0.5, 1),
  legend = c(0.2,0.2) ,legend.title = "", 
  legend.labs = c("high cg02786370 Beta","low cg02786370 Beta"),
  palette = "nejm",
  conf.int = T,
  surv.median.line = "hv"
  )
surv_median(cg10642330_fit_10y)
survdiff(Surv(TenYearSurv, TenDead) ~ cg02786370_beta, data = NMIBC_cg02786370)
```

